nifi-nar-bundles/nifi-py4j-bundle/nifi-py4j-bridge/src/main/java/org/apache/nifi/py4j/PythonProcess.java

```diff
@@ -234,8 +240,18 @@ private Process launchPythonProcess(final int listeningPort, final String authTo
 
         final List<String> commands = new ArrayList<>();
         commands.add(pythonCommand);
+        if (!isUseVirtualEnv()) {
+            // If not using venv, we will not launch a separate virtual environment, so we need to use the -S
+            // flag in order to prevent the Python process from using the installation's site-packages. This provides
+            // proper dependency isolation to the Python process.
+            commands.add("-S");
+        }
 
         String pythonPath = pythonApiDirectory.getAbsolutePath();
+        final String absolutePath = virtualEnvHome.getAbsolutePath();
+        final File dependenciesDir = new File(new File(absolutePath), "NAR-INF/bundled-dependencies");
```

Comment:
Is the intent to add `NAR-INF/bundled-dependencies` in all cases? It seems like it should be conditional on whether the Processor is packaged with dependencies.

Reply:
Good catch.